hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropTest.java

```diff
@@ -1009,53 +1012,157 @@ final Stream<DynamicTest> nftWithRoyaltyFeesPaidByReceiverFails() {
+                            Assertions.assertEquals(currentTreasuryBalance.get(), newTreasuryBalance.get())),
+                    validateChargedUsd("NFT with royalty fee airdrop to treasury", 0.001, 20));
+        }
+
+        @HapiTest
+        @DisplayName("FT with HTS fee with treasury as receiver")
         final Stream<DynamicTest> ftWithRoyaltyFeesPaidByReceiverWithTreasuryReceiver() {
+            // declare treasury account balance variables
+            final AtomicLong currentTreasuryBalance = new AtomicLong();
+            final AtomicLong newTreasuryBalance = new AtomicLong();
             return hapiTest(
-                    cryptoCreate(OWNER),
+                    // set initial treasury balance variable
+                    getAccountBalance(TREASURY_FOR_CUSTOM_FEE_TOKENS).exposingBalanceTo(currentTreasuryBalance::set),
+                    cryptoCreate(OWNER).balance(initialBalance),
                     tokenAssociate(OWNER, FT_WITH_HTS_FIXED_FEE),
                     tokenAssociate(OWNER, DENOM_TOKEN),
                     cryptoTransfer(
                             moving(HTS_FEE, DENOM_TOKEN).between(TREASURY_FOR_CUSTOM_FEE_TOKENS, OWNER),
                             moving(HTS_FEE, FT_WITH_HTS_FIXED_FEE).between(TREASURY_FOR_CUSTOM_FEE_TOKENS, OWNER)),
                     tokenAirdrop(moving(50, FT_WITH_HTS_FIXED_FEE).between(OWNER, TREASURY_FOR_CUSTOM_FEE_TOKENS))
-                            .signedByPayerAnd(TREASURY_FOR_CUSTOM_FEE_TOKENS, OWNER));
+                            .signedByPayerAnd(TREASURY_FOR_CUSTOM_FEE_TOKENS, OWNER)
+                            .payingWith(OWNER)
+                            .via("FT with HTS fee airdrop to treasury"),
+                    // assert owner balance
+                    withOpContext((spec, log) -> {
+                        final var record = getTxnRecord("FT with HTS fee airdrop to treasury");
+                        allRunFor(spec, record);
+                        final var txFee = record.getResponseRecord().getTransactionFee();
+                        // the transaction fee should be charged
+                        final var ownerBalance = getAccountBalance(OWNER)
+                                .hasTinyBars(initialBalance - txFee)
+                                .hasTokenBalance(FT_WITH_HTS_FIXED_FEE, HTS_FEE - 50);
+                        allRunFor(spec, ownerBalance);
+                    }),
+                    // set new treasury balance variable
+                    getAccountBalance(TREASURY_FOR_CUSTOM_FEE_TOKENS)
+                            .exposingBalanceTo(newTreasuryBalance::set)
+                            .hasTokenBalance(FT_WITH_HTS_FIXED_FEE, TOKEN_TOTAL - 2 * HTS_FEE + 50),
+                    // assert treasury balance is not changed
+                    withOpContext((spec, log) ->
+                            Assertions.assertEquals(currentTreasuryBalance.get(), newTreasuryBalance.get())),
```

Comment:
Same here move the getAccountBalance before the withOpContext block and put the assertion in the first block

Reply:
Fixed in the latest commit.